File: motor3d 2.0.ConvertedToC#/VerticeGrafo.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.Linq;
using System.Xml.Linq;
namespace Motor3D.Discreta
{
	public class Vertice
	{
		private List<Arista> mAristas;

		private long mIndice;
		public Arista[] Aristas {
			get { return mAristas.ToArray(); }
		}

		public long Indice {
			get { return mIndice; }
		}

		public long Grado {
			get { return mAristas.Count; }
		}

		public bool GradoPar {
			get { return mAristas.Count % 2 == 0; }
		}

		public bool AristaValida {
			get { return Arista.PrimerVertice == mIndice || Arista.SegundoVertice == mIndice; }
		}

		public Vertice(long Indice)
		{
			if (Indice >= 0) {
				mIndice = Indice;
			} else {
				throw new ExcepcionDiscreta("VERTICE (NEW): El indice no puede ser menor que cero" + Constants.vbNewLine + "Indice=" + Indice.ToString());
			}
		}

		public void EstablecerAristas(params Arista[] Aristas)
		{
			mAristas.Clear();

			foreach (Arista Arista in Aristas) {
				if (AristaValida[Arista])
					mAristas.Add(Arista);
			}
		}

		public void AñadirArista(Arista Arista)
		{
			if (AristaValida[Arista]) {
				mAristas.Add(Arista);
			}
		}
	}
}

File: motor3d 2.0/VerticeGrafo.vb
Namespace Discreta
    Public Class Vertice
        Private mAristas As List(Of Arista)
        Private mIndice As Long

        Public ReadOnly Property Aristas As Arista()
            Get
                Return mAristas.ToArray
            End Get
        End Property

        Public ReadOnly Property Indice As Long
            Get
                Return mIndice
            End Get
        End Property

        Public ReadOnly Property Grado As Long
            Get
                Return mAristas.Count
            End Get
        End Property

        Public ReadOnly Property GradoPar As Boolean
            Get
                Return mAristas.Count Mod 2 = 0
            End Get
        End Property

        Public ReadOnly Property AristaValida(ByVal Arista As Arista) As Boolean
            Get
                Return Arista.PrimerVertice = mIndice OrElse Arista.SegundoVertice = mIndice
            End Get
        End Property

        Public Sub New(ByVal Indice As Long)
            If Indice >= 0 Then
                mIndice = Indice
            Else
                Throw New ExcepcionDiscreta("VERTICE (NEW): El indice no puede ser menor que cero" & vbNewLine & _
                                            "Indice=" & Indice.ToString)
            End If
        End Sub

        Public Sub EstablecerAristas(ByVal ParamArray Aristas() As Arista)
            mAristas.Clear()

            For Each Arista As Arista In Aristas
                If AristaValida(Arista) Then mAristas.Add(Arista)
            Next
        End Sub

        Public Sub AñadirArista(ByVal Arista As Arista)
            If AristaValida(Arista) Then
                mAristas.Add(Arista)
            End If
        End Sub
    End Class
End Namespace

